### ISSUE TITLE
import/require no longer working

### ISSUE THREAD

**Issue by timelyportfolio** [vega/dataflow-api#3]
Not sure if I should file here and not sure what changed but import/require in observablehq no longer working.  See @jheer [examples](https://beta.observablehq.com/@jheer/dataflow-api-examples).

**Comment by jheer** [comment]
Thanks @timelyportfolio! I didn't make any changes or new releases of dataflow-api, so this must have been introduced with a change to Observable.

Perhaps @mbostock has more insight? I'm happy to make changes to the build process here if needed!

**Comment by mbostock** [comment]
I see a couple problems.

The [dataflow-api library](https://unpkg.com/dataflow-api@0.0.1/build/dataflow-api.js) at runtime loads vega-loader as a dependency. For example:

```js
typeof define === 'function' && define.amd ? define(['exports', 'vega-loader'], factory)
```

However, the [dataflow-api package.json](https://github.com/vega/dataflow-api/blob/master/package.json) doesn’t declare this dependency. It declares a different set of dependencies:

```json
"dependencies": {
  "vega-dataflow": "3",
  "vega-transforms": "1",
  "vega-util": "1"
}
```

It appears that the release of vega-loader 4.0 is what broke in Observable. Previously, the [vega-loader 3.1.0 package.json](https://unpkg.com/vega-loader@3.1.0/package.json) declared a main entry of [build/vega-loader.js](https://unpkg.com/vega-loader@3.1.0/build/vega-loader.js). In vega-loader 4.0.0, this switched to [build/vega-loader.node.js](https://unpkg.com/vega-loader@4.0.0/build/vega-loader.node.js). This is a CommonJS module which isn’t compatible with browsers.

I notice that vega-loader 4.0 has a browser entry, as well, but this isn’t supported by [d3-require](https://github.com/d3/d3-require), which is what Observable uses to resolve browser-compatible modules.

Here’s what I recommend:

1. Change the dataflow-api package.json to declare its dependency on vega-loader at whatever version is appropriate (e.g., "3" or "4"). Move the current dependencies to devDependencies or remove them, as appropriate. Release dataflow-api 0.0.2.

2. Change the vega-loader package.json to declare an `unpkg` or `jsdelivr` entry point that points to [build/vega-loader.js](https://unpkg.com/vega-loader@4.0.0/build/vega-loader.js). These entry points are the most specific way of declaring a browser-compatible module.

**Comment by mbostock** [comment]
You can also workaround these problems using require.alias like so:

```js
df = require.alias({"vega-loader": "vega-loader@3"})("dataflow-api")
```

**Comment by jheer** [comment]
Thanks @mbostock! We'll take these suggestions into account.

FWIW, the `vega-loader` dependency is not direct, it is a chained dependency from `vega-dataflow`, which was appropriately version locked in `dataflow-api`.

**Comment by timelyportfolio** [comment]
Thanks @mbostock for the investigation and @jheer for the response in a probably poorly placed issue.  I will make the change on my observable example and reference this issue.

More broadly thanks as always for both of your diligent and brilliant efforts.

**Comment by jheer** [comment]
I just published dataflow-api v0.1.0, which includes vega-loader in the rollup bundle. I've also updated the example notebook on Observable to explicitly use v0.1.0 -- it works! Thanks again everyone.
